paych/lotus/module.go

```diff
@@ -0,0 +1,178 @@
+package lotus
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/api/apistruct"
+	"github.com/filecoin-project/specs-actors/actors/builtin/paych"
+	"github.com/ipfs/go-cid"
+	"github.com/textileio/powergate/ffs"
+)
+
+// Module provides access to the paych api
+type Module struct {
+	api *apistruct.FullNodeStruct
+}
+
+var _ ffs.PaychManager = (*Module)(nil)
+
+// New creates a new paych module
+func New(api *apistruct.FullNodeStruct) *Module {
+	return &Module{
+		api: api,
+	}
+}
+
+// List lists all payment channels involving the specified addresses
+func (m *Module) List(ctx context.Context, addrs ...string) ([]ffs.PaychInfo, error) {
+	filter := make(map[string]struct{}, len(addrs))
+	for _, addr := range addrs {
+		filter[addr] = struct{}{}
+	}
+
+	allAddrs, err := m.api.PaychList(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	chans := make([]<-chan statusResult, len(allAddrs))
+	for i, addr := range allAddrs {
+		chans[i] = m.paychStatus(ctx, addr)
+	}
+
+	resultsCh := make(chan statusResult, len(chans))
+	for _, c := range chans {
+		go func(ch <-chan statusResult) {
+			res := <-ch
+			resultsCh <- res
+		}(c)
+	}
+
+	results := make([]statusResult, len(chans))
+	for i := 0; i < len(chans); i++ {
+		results[i] = <-resultsCh
+	}
```

Comment:
Above, use the wild ways of go async code to concurrently get the pay channel status info for each pay channel address